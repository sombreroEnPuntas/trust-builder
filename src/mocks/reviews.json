[{
        "firstName": "Simon",
        "lastName": "Lock",
        "fullName": "Simon Lock",
        "location": "Kolding",
        "reviewTitle": "Super quality.. I will show here again!",
        "reviewBody": "Super nice quality, fast devilery, good prices. I will shop here again!",
        "starRating": "5"
    },
    {
        "firstName": "Gav",
        "lastName": "",
        "fullName": "Gav",
        "location": "",
        "reviewTitle": "Princely Sum",
        "reviewBody": "A decent local curry house in Faversham, Kent known for its Elvis nights.",
        "starRating": "4"
    },
    {
        "firstName": "Justin",
        "lastName": "Wright",
        "fullName": "Justin Wright",
        "location": "London, GB",
        "reviewTitle": "Good Services",
        "reviewBody": "A decent place to introduce your taste buds to fiery Indian fare",
        "starRating": "3"
    },
    {
        "firstName": "Erika",
        "lastName": "Wolfe",
        "fullName": "Erika Wolfe",
        "location": "Gothenburg, SE",
        "reviewTitle": "Nightmare experience - no product, no communication, no refund; improved by rapid resolution",
        "reviewBody": "In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the chairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the chairs arrive this week as scheduled.\n'), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012. ",
        "starRating": "2"
    },
    {
        "firstName": "Hugo",
        "lastName": "Beja",
        "fullName": "Hugo Beja",
        "location": "Praia Da Barra, PT",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    },
    {
        "firstName": "Simona",
        "lastName": "Locka",
        "fullName": "Simona Locka",
        "location": "Brandenburg",
        "reviewTitle": "Ich hat eine ente",
        "reviewBody": "Die ente frist das hund",
        "starRating": "5"
    },
    {
        "firstName": "Gava",
        "lastName": "",
        "fullName": "Gava",
        "location": "",
        "reviewTitle": "Cesaro Summability",
        "reviewBody": "This was a song I was listening to when I was younger.",
        "starRating": "4"
    },
    {
        "firstName": "Justina",
        "lastName": "Wrighta",
        "fullName": "Justina Wrighta",
        "location": "Leeds, GB",
        "reviewTitle": "Good Lord",
        "reviewBody": "Train be late, ain't no waiting!",
        "starRating": "3"
    },
    {
        "firstName": "Eriko",
        "lastName": "Wolfo",
        "fullName": "Eriko Wolfo",
        "location": "Johanesburg",
        "reviewTitle": "Sunmare experience - yes product, yes communication, yes refund; improved by slow resolution",
        "reviewBody": "In early 2102, I ordered a set of stairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect stairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of stairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the stairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the stairs arrive this week as scheduled.\n'), I still had no stairs, had to go out and buy some other stairs and still had no clue when the Infurn stairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my stairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012.",
        "starRating": "2"
    },
    {
        "firstName": "Huga",
        "lastName": "Bejo",
        "fullName": "Huga Bejo",
        "location": "Fortaleza, BZ",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 stairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of stairs... we paid for 5 PAIRS and received 5 stairs!!! The other 2, the rocket stairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    },
    {
        "firstName": "Simon",
        "lastName": "Lock",
        "fullName": "Simon Lock",
        "location": "Kolding",
        "reviewTitle": "Super quality.. I will show here again!",
        "reviewBody": "Super nice quality, fast devilery, good prices. I will shop here again!",
        "starRating": "5"
    },
    {
        "firstName": "Gav",
        "lastName": "",
        "fullName": "Gav",
        "location": "",
        "reviewTitle": "Princely Sum",
        "reviewBody": "A decent local curry house in Faversham, Kent known for its Elvis nights.",
        "starRating": "4"
    },
    {
        "firstName": "Justin",
        "lastName": "Wright",
        "fullName": "Justin Wright",
        "location": "London, GB",
        "reviewTitle": "Good Services",
        "reviewBody": "A decent place to introduce your taste buds to fiery Indian fare",
        "starRating": "3"
    },
    {
        "firstName": "Erika",
        "lastName": "Wolfe",
        "fullName": "Erika Wolfe",
        "location": "Gothenburg, SE",
        "reviewTitle": "Nightmare experience - no product, no communication, no refund; improved by rapid resolution",
        "reviewBody": "In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.\nWeek after week,\nmonth after month,\nInfurn 's website kept updating the status of the order so that the chairs'\narrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,\nand I had had contact with their customer service in early November(and their only clueless reply was to say, 'We had manufacturing problems; let'\ns hope the chairs arrive this week as scheduled.\n'), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.\nAt some point I\nfinally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They\nfinally offered me a refund - which I accepted on 14 December 2012. ",
        "starRating": "2"
    },
    {
        "firstName": "Hugo",
        "lastName": "Beja",
        "fullName": "Hugo Beja",
        "location": "Praia Da Barra, PT",
        "reviewTitle": "FRAUD",
        "reviewBody": "I've been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it's all... So we bought 'Pairs' of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....\nCommunication ZERO,\nthey simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid\nfor !!",
        "starRating": "1"
    }
]   