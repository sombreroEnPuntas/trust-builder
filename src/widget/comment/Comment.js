import React from 'react';
import PropTypes from 'prop-types';

import './Comment.scss';

const starLine = rating => `${'★'.repeat(rating)}${'☆'.repeat(5 - rating)}`;

const Comment = ({
  review,
}) => (
  <div className="Comment">
    <div className="title">
      <span>{review.reviewTitle}</span>
      <span className="rating">{starLine(review.starRating)}</span>
    </div>
    <div className="sub-title">
      <span>{`${review.fullName}${review.location ? ' - ' : ''}${review.location}`}</span>
    </div>
    <div className="body">
      <p>{review.reviewBody}</p>
    </div>
  </div>
);

Comment.propTypes = {
  review: PropTypes.shape({
    fullName: PropTypes.string,
    location: PropTypes.string,
    reviewBody: PropTypes.string,
    reviewTitle: PropTypes.string,
    starRating: PropTypes.string,
  }).isRequired,
};

Comment.defaultProps = {
};

export default Comment;
