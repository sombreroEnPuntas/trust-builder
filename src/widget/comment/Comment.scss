@import '../styles';

.Comment {
  .title {
    @include text-primary();
    font-weight: bold;
  }

  .rating {
    @include text-stars();
    float: right;
  }

  .sub-title {
    @include text-secondary();
  }

  .body {
    @include text-primary();
  }
}
