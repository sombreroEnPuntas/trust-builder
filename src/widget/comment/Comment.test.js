import React from 'react';
import { shallow } from 'enzyme';

import reviews from '../../mocks/reviews';
import Comment from './Comment';

it('Comment renders without crashing', () => {
  shallow(<Comment review={reviews[0]} />);
});

describe('Comment renders mocks as expected for:', () => {
  reviews.forEach(review => {
    it(review.fullName, () => {
      const wrapper = shallow(<Comment review={review} />);
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
