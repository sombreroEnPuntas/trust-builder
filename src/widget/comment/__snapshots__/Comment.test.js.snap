// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders mocks as expected for:  Erika Wolfe 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"title\\">
    <span>
      Nightmare experience - no product, no communication, no refund; improved by rapid resolution
    </span>
    <span className=\\"rating\\">
      ★★☆☆☆
    </span>
  </div>
  <div className=\\"sub-title\\">
    <span>
      Erika Wolfe - Gothenburg, SE
    </span>
  </div>
  <div className=\\"body\\">
    <p>
      In early 2012, I ordered a set of chairs from Infurn. I thought that by ordering in March, I would have what I needed by November, certainly. I wanted the perfect chairs for my house, and really did not NEED them before the annual Thanksgiving dinner (the only time of year I have a need for a whole lot of chairs at one time). I played it safe, I thought, by ordering so far in advance.
      Week after week,
      month after month,
      Infurn &#39;s website kept updating the status of the order so that the chairs&#39;
      arrival would be further and further in the distant future.Finally when the week of Thanksgiving arrived,
      and I had had contact with their customer service in early November(and their only clueless reply was to say, &#39;We had manufacturing problems; let&#39;
      s hope the chairs arrive this week as scheduled.
      &#39;), I still had no chairs, had to go out and buy some other chairs and still had no clue when the Infurn chairs might arrive.
      At some point I
      finally just requested a refund because Infurn could neither deliver my chairs nor give me a solid date about when I might receive them when I inquired about a delivery date.They
      finally offered me a refund - which I accepted on 14 December 2012. 
    </p>
  </div>
</div>"
`;

exports[`Comment renders mocks as expected for:  Gav 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"title\\">
    <span>
      Princely Sum
    </span>
    <span className=\\"rating\\">
      ★★★★☆
    </span>
  </div>
  <div className=\\"sub-title\\">
    <span>
      Gav
    </span>
  </div>
  <div className=\\"body\\">
    <p>
      A decent local curry house in Faversham, Kent known for its Elvis nights.
    </p>
  </div>
</div>"
`;

exports[`Comment renders mocks as expected for:  Hugo Beja 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"title\\">
    <span>
      FRAUD
    </span>
    <span className=\\"rating\\">
      ★☆☆☆☆
    </span>
  </div>
  <div className=\\"sub-title\\">
    <span>
      Hugo Beja - Praia Da Barra, PT
    </span>
  </div>
  <div className=\\"body\\">
    <p>
      I&#39;ve been patiently waiting for a miracle to happen with our order AU-316084, 12 Chairs!! First the delays, lots and lots of delays with no apparent reason... after they send just 1/2 of the order and say it&#39;s all... So we bought &#39;Pairs&#39; of chairs... we paid for 5 PAIRS and received 5 chairs!!! The other 2, the rocket chairs we never receive....
      Communication ZERO,
      they simply ignore the e - mails...when they did respond they asked us to prove our order to be pairs!!!LOL and just stopped communicating...their website is constantly down...probably to make lose interest and rest your forces to recover what you paid
      for !!
    </p>
  </div>
</div>"
`;

exports[`Comment renders mocks as expected for:  Justin Wright 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"title\\">
    <span>
      Good Services
    </span>
    <span className=\\"rating\\">
      ★★★☆☆
    </span>
  </div>
  <div className=\\"sub-title\\">
    <span>
      Justin Wright - London, GB
    </span>
  </div>
  <div className=\\"body\\">
    <p>
      A decent place to introduce your taste buds to fiery Indian fare
    </p>
  </div>
</div>"
`;

exports[`Comment renders mocks as expected for:  Simon Lock 1`] = `
"<div className=\\"Comment\\">
  <div className=\\"title\\">
    <span>
      Super quality.. I will show here again!
    </span>
    <span className=\\"rating\\">
      ★★★★★
    </span>
  </div>
  <div className=\\"sub-title\\">
    <span>
      Simon Lock - Kolding
    </span>
  </div>
  <div className=\\"body\\">
    <p>
      Super nice quality, fast devilery, good prices. I will shop here again!
    </p>
  </div>
</div>"
`;
